### ISSUE TITLE
Curdling milk bug

### ISSUE THREAD

**Issue by Firestorm01X2** [cataclysmbnteam/Cataclysm-BN#1331]
User bug restored from user screenshot.
# Describe the bug
When I try to put curdling mill into the fermenting vat I get errors and the curdles disappear

I've also tried putting various musts (wine/mead etc) into the vat and they worked fine. Its just the curdles that break the vat

# Steps To Reproduce
I craft regular curdle milk from cows milk. The I put sime in containers (usually into a plastic jerrycan) amd tjem I put the container info fermentation vat. Then when I examine on the vat I get error screen "fvat_ful was empty" and curdle mix vanushes

# Expected behavior
I expected some fermented curdles from from the vat for cheese making. But vat doesn't give me any. I only get error messages instead (see the log from them)

# Screenshots
https://i.ibb.co/mzhHpVs/spam-again.jpg

# Versions and configuration

# Additional context

**Comment by cbmcat2** [comment]
Here is some additional info about the bug:
Debug log: https://ufile.io/fc5hf75l
Save file: https://ufile.io/3e7mcjov

**Comment by chaosvolt** [comment]
Is it possibly because, if I recall, curdling milk is the only fermentable item with `spoils_in` set?

**Comment by cbmcat2** [comment]
> if I recall, curdling milk is the only fermentable item with `spoils_in` set?

Sounds like that could be something. I always craft with unrotten foods but maybe the vat no longer knows how to account for such variables.

**Comment by Coolthulhu** [comment]
Looks like vat needs `"DONT_REMOVE_ROTTEN"` flag. If it has it, then it is not working.

**Comment by chaosvolt** [comment]
Empty and full vats have it, yeah:
1. https://github.com/cataclysmbnteam/Cataclysm-BN/blob/upload/data/json/furniture_and_terrain/furniture-tools.json#L781
2. https://github.com/cataclysmbnteam/Cataclysm-BN/blob/upload/data/json/furniture_and_terrain/furniture-tools.json#L816

**Comment by chaosvolt** [comment]
Another oddity I've noticed: `"brewable": { "time": "4 hours", "results": [ "milk_curdled" ] }`

![image](https://user-images.githubusercontent.com/11582235/154627982-9d4d0060-0380-40b8-b6b4-66efd80b2d1f.png)

Might warrant a separate issue report if all items are affected by this?

**Comment by Coolthulhu** [comment]
@chaosvolt 
```C++
time_duration item::brewing_time() const
{
    return is_brewable() ? type->brewable->time * calendar::season_ratio() : 0_turns;
}
```

It scales with season length for some reason. I'd just get rid of the scaling.

**Comment by chaosvolt** [comment]
Weird. Seems like more fodder for a PR then. So that just leaves sorting out what might be causing issues with curdling milk.

Part of my planned changes to cheesemaking, as mentioned in https://github.com/cataclysmbnteam/Cataclysm-BN/issues/883, would involve altering the fermentation time for 4 hours down to 1 hour, as part of some changes to make it so that making a large batch of cheese the "hard way" will reliably take less time than making the same amount of cheese the other way.

I've got some JSON set aside for tinkering with to tackle that idea after I've got my fork properly re-synced.
